src/vm/class.cpp

```diff
@@ -3757,15 +3754,15 @@ VOID EEClassLayoutInfo::CollectLayoutFieldMetadataThrowing(
 
             UINT numChildCTMFields = pEEClassLayoutInfoOut->m_numCTMFields - pParentLayoutInfo->m_numCTMFields;
 
-            BYTE *pParentCTMFieldSrcArray = (BYTE*)pParentLayoutInfo->GetFieldMarshalers();
-            BYTE *pParentCTMFieldDestArray = ((BYTE*)pEEClassLayoutInfoOut->GetFieldMarshalers()) + MAXFIELDMARSHALERSIZE*numChildCTMFields;
+            NativeFieldDescriptor *pParentCTMFieldSrcArray = pParentLayoutInfo->GetNativeFieldDescriptors();
+            NativeFieldDescriptor *pParentCTMFieldDestArray = pEEClassLayoutInfoOut->GetNativeFieldDescriptors() + numChildCTMFields;
 
             for (UINT parentCTMFieldIndex = 0; parentCTMFieldIndex < pParentLayoutInfo->m_numCTMFields; parentCTMFieldIndex++)
```

Comment:
This is basically a call to `memmove()` and would probably be much more efficient. If you can guarentee there is no overlap feel free to use `memcpy()`.

Reply:
We need to call the copy constructors here to ensure we have the correct fixups for the FieldDesc (and MethodTable) in the NativeFieldDescriptor. In the CoreCLR build they're just `PlainPointer`s so it would be fine to just do a memcpy, but in crossgen they're relative fixups so we need to do the copy.

Reply:
Boo. Okay.